cylc/flow/scripts/install.py

```diff
@@ -171,14 +191,76 @@ def get_source_location(path: Optional[str]) -> Path:
     return search_install_source_dirs(expanded_path)
 
 
+async def scan(wf_name: str, ping: bool = True) -> None:
+    """Print any instances of wf_name that are already active."""
+    opts = Values({
+        'name': [f'{wf_name}/*'],
```

Comment:
Might want to restrict this to numbered runs?

```suggestion
        'name': [f'{wf_name}/run*'],
```

I think this approach will actually list all workflows on the filesystem, then filter by name. There is a way to restrict scanning to just the "install dir" we are interested in which should be a good bit faster, will take a look.

Reply:
Regarding your suggested change, I was trying to catch named runs as well. Is there some reason not to do that?

> There is a way to restrict scanning to just the "install dir" we are interested in which should be a good bit faster, will take a look.

Did you take a look?

Reply:
> Is there some reason not to do that?

I expect having multiple named runs of a workflow running simultaneously to be a likely pattern.

E.G. our post-processing is defined in a single workflow which is generic, when it is installed, it becomes specialised to it's usage via Jinja2. So one workflows, multiple concurrent named runs.

But meh, maybe leave it as is and adapt later if desired.

> Did you take a look?

Use the `scan_dir` argument:

https://github.com/cylc/cylc-flow/blob/79721142e872fb6fde97a79ef7a6f05e9332384c/cylc/flow/network/scan.py#L188-L215

Here's an example of an integration test which uses this argument to scan a single directory:

https://github.com/cylc/cylc-flow/blob/79721142e872fb6fde97a79ef7a6f05e9332384c/tests/integration/test_scan.py#L319-L333

Reply:
Scanning all workflows can get very slow (especially on NFS) so this is quite important for us.

Reply:
> 13 users with >500 with the prize going to a user with a whopping 2021 workflows. 

:exploding_head: 

Anyhow, great, I'll tweak it to restrict the directory...

Reply:
> I expect having multiple named runs of a workflow running simultaneously to be a likely pattern.

Me too, which is why I was trying to alert the user to those too.  So I still don't quite understand your point.  Maybe it was that WARNING (which I've backed out of) is too strong when those runs are deliberate?  But the same goes for `runN` runs, in principle.

Reply:
scan now restricted to the one workflow